src/cpp/library/vector

```diff
@@ -189,7 +189,8 @@ public:
 
   vector(const vector &x) : _size(0), _capacity(1)
   {
-    buf = new T[x._size];
+    buf = static_cast<T*>(malloc(sizeof(T) * x._size));
```

Comment:
Aren't you creating a memleak here? Maybe using something like __ESBMC_alloca or add a free in a destructor (is there one available?)